test/Integration.cpp

```diff
@@ -383,6 +383,867 @@ TEST_F(ZNCTest, InvalidConfigInChan) {
 	znc->ShouldFinishItself(1);
 }
 
+TEST_F(ZNCTest, ControlpanelModule) {
+	auto znc = Run();Z;
+	auto ircd = ConnectIRCd();Z;
+	auto client = LoginClient();Z;
+
+	const QString request = "PRIVMSG *controlpanel :";
+	const QByteArray response = ":*controlpanel!znc@znc.in PRIVMSG nick :";
+
+	// TODO: Figure out how to check for "HAVE_ICU" to test encoding.
+
+	// 1. Test everything on "user".
+	// 2. Test everything on another user "KindOne" from "user".
+	// 3. Test one thing on "foobar" as the user does not exist from "user".
+
+	// Add myself so I can test other things along with "user".
+	client.Write(request + "AddUser KindOne hunter2");
+	client.ReadUntil(response + "User [KindOne] added!");Z;
+
+	client.Write(request + "AddUser KindOne hunter2");
+	client.ReadUntil(response + "Error: User [KindOne] already exists!");Z;
+
+	client.Write(request + "AddCTCP user VERSION Test");
+	client.ReadUntil(response + "Added!");Z;
+
+	client.Write("PRIVMSG nick :\x01VERSION\x01");
+	ircd.Write(":NOTICE nick :mIRC 7.42");Z;
```

Comment:
This doesn't test anything, be it "Test" or "mIRC" or something else. If it did test something, the test would probably fail. Good test should fail when the thing which it tests behaves differently from what the test expects.

This whole test now tests only interaction with user, but it doesn't test that the settings are actually applied and used somewhere. Please add a TODO to add that.
Then a real test that CTCP VERSION does what it's supposed to do can be added later.